sys/include/net/netopt.h

```diff
@@ -574,6 +577,84 @@ typedef enum {
      */
     NETOPT_PHY_BUSY,
 
+    /**
+     * @brief   (uint8_t*) LoRaWAN device EUI (8 bytes length)
+     */
+    NETOPT_LORAWAN_DEVEUI,
+
+    /**
+     * @brief   (uint8_t*) LoRaWAN application EUI (8 bytes length)
+     */
+    NETOPT_LORAWAN_APPEUI,
+
+    /**
+     * @brief   (uint8_t*) LoRaWAN application key (16 bytes length)
+     */
+    NETOPT_LORAWAN_APPKEY,
+
+    /**
+     * @brief   (uint8_t*) LoRaWAN network session key (16 bytes length)
+     */
+    NETOPT_LORAWAN_NWKSKEY,
+    /**
+     * @brief   (uint8_t*) LoRaWAN application session key (16 bytes length)
+     */
+    NETOPT_LORAWAN_APPSKEY,
+
+    /**
+     * @brief   (@ref netopt_enable_t) LoRaWAN join state
```

Comment:
I think  NETOPT_LINK_CONNECTED suits perfect for this. I'm already using it for #11022.